tools/tests/package-tests.js

```diff
@@ -1952,3 +1958,637 @@ selftest.define("show package w/many versions",
+  testShowPackageVersion(s, baseVersionInfo);
+
+  // Some formatting in the text.
+  var excerpt =
+        "Here is a code sample:" + "\n\n" +
+        "```foobar\nand foobar```";
+  readme =
+    "Heading" + "\n" +
+    "========" + "\n" + "\n" +
+    excerpt + "\n\n" +
+    "# Subheading" + "\n" + "\n" +
+    "## Another subheading" + "\n" + "\n" +
+    "You should not see this line!";
+  s.write("README.md", readme);
+  testShowPackage(
+    s, name, _.extend({ description: excerpt }, basePackageInfo));
+  testShowPackageVersion(
+    s, _.extend({ description: excerpt }, baseVersionInfo));
+
+  // Some formatting in the text that we are going to skip.
+  readme =
+    "Heading" + "\n" +
+    "========" + "\n" + "\n" +
+    "# Subheading" + "\n" + "\n" +
+    "![skip this image!] \n" +
+    excerpt + "\n\n" +
+    "## Another subheading" + "\n" + "\n" +
+    "You should not see this line!";
+  s.write("README.md", readme);
+  testShowPackage(
+    s, name, _.extend({ description: excerpt }, basePackageInfo));
+  testShowPackageVersion(
+    s, _.extend({ description: excerpt }, baseVersionInfo));
+
+  // Now, let's try different file specifications for the documentation.
+  var git = "https:ilovegit.git";
+  var summary = "Test summary";
+  var staging = s.read("package-customizable.js");
+  staging = staging.replace(/~version~/g, "1.0.0");
+  staging = staging.replace(/~git~/g, git);
+  staging = staging.replace(/~summary~/g, summary);
+
+  // If we specify null, we have no docs.
+  s.write("package.js", staging.replace(/~documentation~/g, "null"));
+  baseVersionInfo = {
+    summary: summary,
+    git: git,
+    packageName: name,
+    version: "1.0.0",
+    directory: packageDir
+  };
+  basePackageInfo = {
+    git: git,
+    summary: summary,
+    versions: [{ version: "1.0.0", directory: packageDir }]
+  };
+  testShowPackageVersion(s, baseVersionInfo);
+  testShowPackage(s, name, basePackageInfo);
+
+  // If we specify a different file, read that file.
+  s.write("package.js",
+          staging.replace(/~documentation~/g, "'Meteor-Readme.md'"));
+  readme = "A special Readme, just for Meteor.";
+  s.write("Meteor-Readme.md", "Title\n==\n" + readme);
+  testShowPackageVersion(s,
+    _.extend({ description: readme }, baseVersionInfo));
+  testShowPackage(s, name,
+    _.extend({ description: readme }, basePackageInfo));
+
+  // If we specify a non-existent file, tell us.
+  s.write("package.js",
+          staging.replace(/~documentation~/g, "'NOTHING'"));
+  var run = s.run("show", name);
+  run.matchErr("Documentation not found");
+  run.expectExit(1);
+  run = s.run("show", name + "@1.0.0");
+  run.matchErr("Documentation not found");
+  run.expectExit(1);
+});
+
+// Show publication with different types of readme files.
+selftest.define("show server readme",
+  ['net', 'test-package-server', 'slow'], function () {
+
+  var s = new Sandbox();
+  s.set("METEOR_TEST_TMP", files.mkdtemp());
```

Comment:
Looks like it is used by one specific test introduced in 84e1d5f7cd with a very tight scope between write and read.  And I think it may have just gotten copied into a bunch of other tests?